src/test/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectivesSpec.scala

```diff
@@ -0,0 +1,133 @@
+    }
+  }
+
+  it should "fail if PET account is not found" in {
+    forAll(genOAuth2BearerToken, genPetEmail){
+      (token: OAuth2BearerToken, email: WorkbenchEmail) =>
+        val directoryDAO = new MockDirectoryDAO()
+        val googleSubjectId = GoogleSubjectId(genRandom(System.currentTimeMillis()))
+        Try(getUserInfo(token, googleSubjectId, email, 10L, directoryDAO).futureValue) === (Failure(new WorkbenchException("no owner found for this PET account")))
+    }
+  }
+
+  it should "fail if a regular user email is provided, but subjectId is not for a regular user" in {
+    forAll(genServiceAccountSubjectId, genOAuth2BearerToken, genNonPetEmail){
+      (serviceSubjectId: ServiceAccountSubjectId, token: OAuth2BearerToken, email: WorkbenchEmail) =>
+        val directoryDAO = new MockDirectoryDAO()
+        val gSid = GoogleSubjectId(serviceSubjectId.value)
+        val uid = genWorkbenchUserId(System.currentTimeMillis())
+        directoryDAO.createPetServiceAccount(PetServiceAccount(PetServiceAccountId(uid, GoogleProject("")), ServiceAccount(serviceSubjectId, email, ServiceAccountDisplayName("")))).futureValue
+        Try(getUserInfo(token, gSid, email, 10L, directoryDAO).futureValue) === (Failure(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"subjectId $gSid is not a WorkbenchUser"))))
+    }
+  }
+
+  it should "be able to create CreateWorkbenchUserAPI object" in {
+    forAll{
+      (email: WorkbenchEmail) =>
+        val googleSubjectId = GoogleSubjectId(genRandom(System.currentTimeMillis()))
+        val res = createWorkbenchUserAPI(googleSubjectId, email).futureValue
+        res.googleSubjectId should be (googleSubjectId)
+        res.email should be (email)
+    }
+  }
+
+  it should "fail to create CreateWorkbenchUserAPI object for PET" in {
+    forAll(genPetEmail){
+      (email: WorkbenchEmail) =>
+        val googleSubjectId = GoogleSubjectId(genRandom(System.currentTimeMillis()))
+        Try(createWorkbenchUserAPI(googleSubjectId, email).futureValue) === (Failure(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"owner for PET ${googleSubjectId} already exists. Use a different endpoint for creating PET account"))))
+    }
+  }
+
+  it should "fail if expiresIn is in illegal format" in {
+    forAll(genUserInfoHeadersWithInvalidExpiresIn){
+      headers: List[RawHeader] =>
+        Get("/").withHeaders(headers) ~> directives.requireUserInfo(x => complete(x.toString)) ~> check {
+          Try(responseAs[String])  === (Failure(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"expiresIn ${headers(3).value} can't be converted to Long"))))
+        }
+    }
+  }
+
+  it should "fail if required headers are missing" in {
+    Get("/") ~> directives.requireUserInfo(x => complete(x.toString)) ~> check {
+      Try(responseAs[String])  === (Success(MissingHeaderRejection(accessTokenHeader)))
+    }
+  }
+}
```

Comment:
I tried to add success tests. But it's not easy to test just `Directives` with success case since inside `check` block, it's probably referencing some object that's not exposed, hence I'm getting `NPE`.

All tests that are using `SamRoutes` rely on this code to be in happy path, so I don't feel like spending more time adding successful case for just `requireUserInfo` .